main/src/ca/uwaterloo/flix/language/phase/CodegenHelper.scala

```diff
@@ -0,0 +1,627 @@
+      case Expression.LoadInt16(b, o) => Set.empty
+      case Expression.LoadInt32(b, o) => Set.empty
+      case Expression.StoreBool(b, o, v) => Set.empty
+      case Expression.StoreInt8(b, o, v) => Set.empty
+      case Expression.StoreInt16(b, o, v) => Set.empty
+      case Expression.StoreInt32(b, o, v) => Set.empty
+      case Expression.Var(sym, tpe, loc) => Set.empty
+      case Expression.Ref(name, tpe, loc) => Set.empty
+      case Expression.MkClosureRef(ref, freeVars, tpe, loc) => Set(tpe)
+      case Expression.ApplyRef(name, args, tpe, loc) => args.flatMap(visit).toSet
+      case Expression.ApplyTail(name, formals, actuals, tpe, loc) => actuals.flatMap(visit).toSet
+      case Expression.ApplyHook(hook, args, tpe, loc) => args.flatMap(visit).toSet
+      case Expression.ApplyClosure(exp, args, tpe, loc) => visit(exp) ++ args.flatMap(visit)
+      case Expression.Unary(op, exp, tpe, loc) => visit(exp)
+      case Expression.Binary(op, exp1, exp2, tpe, loc) => visit(exp1) ++ visit(exp2)
+      case Expression.IfThenElse(exp1, exp2, exp3, tpe, loc) => visit(exp1) ++ visit(exp2) ++ visit(exp3)
+      case Expression.Let(sym, exp1, exp2, tpe, loc) => visit(exp1) ++ visit(exp2)
+      case Expression.LetRec(sym, exp1, exp2, tpe, loc) => visit(exp1) ++ visit(exp2)
+      case Expression.Is(sym, tag, exp, loc) => visit(exp)
+      case Expression.Tag(enum, tag, exp, tpe, loc) => visit(exp)
+      case Expression.Untag(sym, tag, exp, tpe, loc) => visit(exp)
+      case Expression.Index(base, offset, tpe, loc) => visit(base)
+      case Expression.Tuple(elms, tpe, loc) => elms.flatMap(visit).toSet
+      case Expression.Reference(exp, tpe, loc) => ??? // TODO
+      case Expression.Dereference(exp, tpe, loc) => ??? // TODO
+      case Expression.Assignment(exp1, exp2, tpe, loc) => ??? // TODO
+      case Expression.Existential(params, exp, loc) =>
+        ???
+      case Expression.Universal(params, exp, loc) =>
+        ???
+      case Expression.NativeConstructor(constructor, args, tpe, loc) => args.flatMap(visit).toSet
+      case Expression.NativeField(field, tpe, loc) => Set.empty
+      case Expression.NativeMethod(method, args, tpe, loc) => args.flatMap(visit).toSet
+      case Expression.UserError(tpe, loc) => Set.empty
+      case Expression.MatchError(tpe, loc) => Set.empty
+      case Expression.SwitchError(tpe, loc) => Set.empty
+    }
+
+    val types = consts.flatMap(x => visit(x.exp)).toSet
+    types.map { t =>
+      val name = Symbol.freshVarSym("FnItf").toString
+      val prefix = FlixClassName(List("ca", "uwaterloo", "flix", "runtime", name))
+      t -> prefix
+    }.toMap
+  }
+
+
+  /**
+    * Find any appearance of an enum case in the give expression `e`
+    * @param e expression
+    * @return List of enum cases in the expression, each element of the list is of form (EnumType, (CaseName, EnumFieldType))
+    */
+  def findEnumCases(e: Expression): List[(Type, (String, Type))] = e match {
```

Comment:
This can't go there, it is needed in CodeGen to create a map from enum types to QualName.